.eslintrc.js

```diff
@@ -46,25 +46,29 @@ module.exports = {
 				selector: 'ImportDeclaration[source.value=/^components$/]',
 				message: 'Use @wordpress/components as import path instead.',
 			},
+			{
+				selector: 'ImportDeclaration[source.value=/^data$/]',
+				message: 'Use @wordpress/data as import path instead.',
+			},
 			{
 				selector: 'ImportDeclaration[source.value=/^date$/]',
 				message: 'Use @wordpress/date as import path instead.',
 			},
 			{
-				selector: 'ImportDeclaration[source.value=/^editor$/]',
-				message: 'Use @wordpress/editor as import path instead.',
+				selector: 'ImportDeclaration[source.value=/^deprecated/]',
```

Comment:
Nit: This pattern should be `/^deprecated$/` (otherwise matching e.g. `deprecatedfoo`)

Reply:
I noticed it too late, it is included together with changes planned for `data` package. Sorry about that :)